pebblo/entity_classifier/entity_classifier.py

```diff
@@ -18,98 +18,91 @@ class EntityClassifier:
-            response = self.anomyze_response(analyzer_results, input_text)
-            logger.debug(f"Presidio Entity Classifier Response: {response}")
-            entities, total_count = get_entities(Entities, response)
-            logger.debug(f"Presidio Entity Classifier Finished {entities}")
-            logger.debug(f"Entity Total count. {total_count}")
-            return entities, total_count
-        except Exception as e:
-            logger.error(f"Presidio Entity Classifier Failed, Exception: {e}")
-            return entities, total_count
-
-    def presidio_secret_classifier(self, input_text):
-        """
-        Find regex patterns in the given input data.
-
-        Parameters:
-        - input_data (str): The input string to search for regex patterns.
-
-        Returns:
-        dict: A dictionary containing key associated with regex patterns and their counts in the input data.
+        And also returns plain input text as anonymized text output
+        :param all_entities: conditional param to check whether we need to anonymize all entities
+        :param input_text: Input string / document snippet
+        :return: entities: containing the entity group Name as key and its count as value.
+                 total_count: Total count of entity groupsInput text in anonymized form.
+                 anonymized_text: Input text in anonymized form.
         Example:
-        {
-            'AWS API ID': 3, # SecretKey: Count of Occurrences of secret key
-            'AWS API Secret': 1
-        },
-        total_count: 4 # total count of all SecretKeys Occurrences
+
+        input_text = " My SSN is 222-85-4836.
+        ITIN number 993-77 0690
+        And AWS Access Key is: AKIAQIPT4PDORIRTV6PH."
+        response:
+        entities = {'AWS Access Key': 1, 'US ITIN': 1, 'US SSN': 1}
+        total_count = 3
+        anonymized_text = "My SSN is <US_SSN>.
+            ITIN number <US_ITIN>
+            And AWS Access Key is: <AWS_ACCESS_KEY>"
         """
-        secret_entities = {}
+        entities = {}
         total_count = 0
+        anonymized_text = ""
         try:
-            logger.debug("Presidio Secret Entity Classifier Started.")
+            logger.debug("Presidio Entity Classifier and Anonymizer Started.")
             logger.debug(f"Data Input: {input_text}")
 
-            # Adding custom analyzer
-            custom_registry = add_custom_regex_analyzer_registry()
-            self.analyzer = AnalyzerEngine(
-                registry=custom_registry,
-                context_aware_enhancer=LemmaContextAwareEnhancer(
-                    context_similarity_factor=float(
-                        ConfidenceScore.EntityContextSimilarityFactor.value
-                    ),
-                    min_score_with_context_similarity=float(
-                        ConfidenceScore.EntityMinScoreWithContext.value
-                    ),
-                ),
+            self.custom_analyze()
```

Comment:
Does it have any impact because of combining entities and secrets?

Reply:
Here we are combining predefined as well as custom recognizers